sdk/core/core-client/src/utils.ts

```diff
@@ -122,14 +122,12 @@ export function flattenResponse(
 ): unknown {
   const parsedHeaders = fullResponse.parsedHeaders;
 
-  /**
-   * If body is not asked for, only response headers are returned. If the response
-   * has a body anyway, that body must be ignored.
-   */
   if (fullResponse.request.method === "HEAD") {
-    return parsedHeaders;
+    return {
+      ...parsedHeaders,
+      body: fullResponse.parsedBody
```

Comment:
can we have a comment like on 199 of deserialization policy that talks about this being a boolean based on success/failure of the request?

Reply:
You can safely remove this whole check if we're going to accept the PR since your current approach has a problem which is assuming the body of all head requests to be a primitive value which is not necessarily true in the broad case.

Reply:
@sarangan12, I still believe you're better off removing the entire check, the boolean body will be wrapped by the standard logic down there.

Reply:
@deyaaeldeen I agree and believe we could remove the entire check. But, I also believe it would not be safe to remove it at this point since I plan to release it in the next few hours.

Reply:
Not being safe because there are not tests for the behavior you expect? if so, please open an issue to add those tests later and to remove the check.